tests/components/heos/test_config_flow.py

```diff
@@ -47,36 +50,42 @@
 
 async def test_create_entry_when_host_valid(hass, controller):
     """Test result type is create entry when host is valid."""
-    flow = HeosFlowHandler()
-    flow.hass = hass
     data = {CONF_HOST: "127.0.0.1"}
-    result = await flow.async_step_user(data)
-    assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
-    assert result["title"] == "Controller (127.0.0.1)"
-    assert result["data"] == data
-    assert controller.connect.call_count == 1
-    assert controller.disconnect.call_count == 1
+    with patch(
+        "homeassistant.components.hue.async_setup_entry", return_value=mock_coro(True)
+    ):
+        result = await hass.config_entries.flow.async_init(
+            heos.DOMAIN, context={"source": "user"}, data=data
+        )
+        assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
+        assert result["title"] == "Controller (127.0.0.1)"
+        assert result["data"] == data
+        assert controller.connect.call_count == 1
+        assert controller.disconnect.call_count == 1
 
 
 async def test_create_entry_when_friendly_name_valid(hass, controller):
     """Test result type is create entry when friendly name is valid."""
     hass.data[DATA_DISCOVERED_HOSTS] = {"Office (127.0.0.1)": "127.0.0.1"}
-    flow = HeosFlowHandler()
-    flow.hass = hass
     data = {CONF_HOST: "Office (127.0.0.1)"}
-    result = await flow.async_step_user(data)
-    assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
-    assert result["title"] == "Controller (127.0.0.1)"
-    assert result["data"] == {CONF_HOST: "127.0.0.1"}
-    assert controller.connect.call_count == 1
-    assert controller.disconnect.call_count == 1
-    assert DATA_DISCOVERED_HOSTS not in hass.data
+    with patch(
+        "homeassistant.components.hue.async_setup_entry", return_value=mock_coro(True)
```

Comment:
`return_value=True`

Reply:
Thank you for your review again!
Finally, all tests are passed.